### PR TITLE
py-fn: new package

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-fn/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class PyFn(PythonPackage):
+    """Functional programming in Python: implementation of missing features
+    to enjoy FP."""
+
+    homepage = "https://github.com/kachayev/fn.py"
+    url      = "https://github.com/kachayev/fn.py"
+
+    version('0.4.3', git='https://github.com/kachayev/fn.py', commit='df53f5d')
+
+    depends_on('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fn/package.py]: This line is already present in the `PythonPackage` base class, so you don't need it here.

File: var/spack/repos/builtin/packages/py-fn/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('py-setuptools')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fn/package.py]: `type='build'`

File: var/spack/repos/builtin/packages/py-fn/package.py (hunk 1 above)
Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-fn/package.py]: done.

File: var/spack/repos/builtin/packages/py-fn/package.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class PyFn(PythonPackage):
+    """Functional programming in Python: implementation of missing features
+    to enjoy FP."""
+
+    homepage = "https://github.com/kachayev/fn.py"
+    url      = "https://github.com/kachayev/fn.py"
+
+    version('0.4.3', git='https://github.com/kachayev/fn.py', commit='df53f5d')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fn/package.py]: It looks like PyPI already has version 0.5.2 if you want to directly download from there.

Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-fn/package.py]: Changed, thanks.

File: var/spack/repos/builtin/packages/py-fn/package.py (hunk 2 above)
Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-fn/package.py]: done.

File: var/spack/repos/builtin/packages/py-fn/package.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyFn(PythonPackage):
+    """Functional programming in Python: implementation of missing features
+    to enjoy FP."""
+
+    homepage = "https://github.com/fnpy/fn.py"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fn/package.py]: I always feel awkward about downloading from a fork, but the original project looks like it's been dead for a while and it seems like this fork has upload privileges on PyPI, so I think it's fine.

**Pull request merged**